rell/src/user/asset.rell

```diff
@@ -0,0 +1,62 @@
+import ft3_core: lib.ft3.core;
+
+query get_user_vibes(name) {
+	val user = user@{ name };
+	val balance = get_balance(user);
+	
+	if (balance == null)
+		return 0;
+	else
+		return balance.amount;
+}
+
+function reward_asset(user, amount: integer) {
+	val account_balance = get_balance(user);
+	
+	val modified_amount = calculate_asset_with_modifier(amount);
+	
+	if (account_balance == null) {
+		create_balance(user, modified_amount);
+		return;
+	}
+	
+	update account_balance(amount = account_balance.amount+modified_amount);
+}
+
+function get_balance(user): ft3_core.balance? {
+	return ft3_core.balance@?{ 
+		user.account,
+		.asset.name == asset_name()
+	};
+}
+
+function create_balance(user, amount: integer) {
+	var asset = ft3_core.asset@?{ .name == asset_name() };
+	
+	if (asset == null) {
+		asset = ft3_core.register_asset(asset_name(), chain_context.blockchain_rid);
+	}
+	
+	create ft3_core.balance(user.account, asset, amount);
+}
+
+function asset_name(): text {
+	return "Vibe";
+}
+
+function calculate_asset_with_modifier(amount: integer): integer {
+	val block = op_context.block_height;
+	
+	return when {
+		block < 10000000 -> amount * 10;
```

Comment:
```suggestion
    val stopMultiplier = 90000000;
    
    val usersWithBalance =  ft3_core.balance@*{.asset.name}.size();
    
		return if (usersWithBalance < stopMultiplier) amount else amount * stopMultiplier / userWithBalance;
```

Probably use the blog as multiplier is not the best since you can't guess the curve (will people join in a week, a month or a year?).
Maybe you could multiply by total number of people with amount? See example.

In this example I assume that 50 ppl will receive balance before allowing to give rate (otherwise you give a lot of tokens to the first topics rewarded)